src/components/application_manager/src/application_impl.cc

```diff
@@ -442,11 +443,20 @@ void ApplicationImpl::StopStreaming(
 
   SuspendStreaming(service_type);
 
-  if (service_type == ServiceType::kMobileNav && video_streaming_approved()) {
-    StopNaviStreaming();
-  } else if (service_type == ServiceType::kAudio &&
-             audio_streaming_approved()) {
-    StopAudioStreaming();
+  if (!helpers::Compare<ServiceType, helpers::EQ, helpers::ONE>(
+          service_type, ServiceType::kMobileNav, ServiceType::kAudio)) {
+      return;
+  }
+
+  const bool as_approved = audio_streaming_approved();
```

Comment:
@AByzhynar 
I would prefer direct calls to the methods as their names are clear as opposed to variables